include/hipSYCL/sycl/device.hpp

```diff
@@ -313,11 +313,11 @@ HIPSYCL_SPECIALIZE_GET_INFO(device, max_work_item_dimensions)
 HIPSYCL_SPECIALIZE_GET_INFO(device, max_work_item_sizes)
 {
   std::size_t size0 = static_cast<std::size_t>(get_rt_device()->get_property(
-      rt::device_uint_property::max_global_size0));
+      rt::device_uint_property::max_group_size0));
   std::size_t size1 = static_cast<std::size_t>(get_rt_device()->get_property(
-      rt::device_uint_property::max_global_size1));
+      rt::device_uint_property::max_group_size1));
   std::size_t size2 = static_cast<std::size_t>(get_rt_device()->get_property(
-      rt::device_uint_property::max_global_size2));
+      rt::device_uint_property::max_group_size2));
   return id<3>{size0, size1, size2};
```

Comment:
Pre-existing problem: given how we "flip" the launch dimensions for CUDA, HIP (https://github.com/illuhad/hipSYCL/blob/f3177b86c7731c27805dfff51c1efa2e7b96cb1f/include/hipSYCL/glue/generic/hiplike/hiplike_kernel_launcher.hpp#L364), and ZE (https://github.com/illuhad/hipSYCL/blob/f3177b86c7731c27805dfff51c1efa2e7b96cb1f/include/hipSYCL/glue/ze/ze_kernel_launcher.hpp#L375), should we now flip the order here as well?

Reply:
I'd say, no. Index flipping is a problem of the kernel headers and the glue code, not of the runtime library which is currently completely unaware that something like this happens. Also, even if we wanted to, we couldn't expose the right flipping here because the correct flip depends on the dimensionality of the query. I.e. `max_size[0]` would depend on whether we are asking for a 1D, 2D, or 3D kernel.

The reason why SYCL 2020 changed the device descriptors to structs instead of enums was precisely to carry the information about the dimensionality. Because I don't want a templated query interface for the runtime, this is another reason why it probably should be handled by the headers. We will have to expose though whether a backend has vectorization across first index as a runtime property so that the device queries in the headers can do the right thing.

Reply:
@nmnobre To clarify: Eventually those queries should also include the flip in hipSYCL. I was just talking about *where* it would eventually need to go. This requires distinguishing between the SYCL headers, where `get_info<>()` lives and the actual hipSYCL runtime library `libhipSYCL-rt` and its backend plugins. I was just saying that in the hipSYCL architecture, that flip should probably be implemented inside `get_info<>`, not the runtime library itself.

For this we probably need to move to the SYCL 2020 notion where structs are used instead of enums for those queries (#774). In SYCL 2020, `max_work_item_sizes<Dim>` is templated with the dimensionality so that the flip can be included correctly.

If DPC++ also flips without knowing the dimensionality, I would consider it confusing and not part of SYCL 2020 since this query does not exist anymore without template parameter for the dimension.
https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_device_information_descriptors

Reply:
@illuhad, yes, you're right; no way to fix it currently.

Reply:
> If DPC++ also flips without knowing the dimensionality, I would consider it confusing and not part of SYCL 2020 since this query does not exist anymore without template parameter for the dimension. https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_device_information_descriptors

I've checked and, my bad, they have in fact transitioned to use structs for the device descriptors a couple months back and they do require the template parameter as per SYCL 2020. I was using an old version which still used the enums and no dimension specification. For clarity, the flip happened then and it still does now. :-)